src/commands/sarif/upload.ts

```diff
@@ -0,0 +1,199 @@
+import fs from 'fs'
+import path from 'path'
+
+import Ajv from 'ajv'
+import addFormats from 'ajv-formats'
+import chalk from 'chalk'
+import {Command} from 'clipanion'
+import glob from 'glob'
+import asyncPool from 'tiny-async-pool'
+
+import {getCISpanTags} from '../../helpers/ci'
+import {getGitMetadata} from '../../helpers/git/format-git-span-data'
+import {SpanTags} from '../../helpers/interfaces'
+import {retryRequest} from '../../helpers/retry'
+import {parseTags} from '../../helpers/tags'
+import {getUserGitSpanTags} from '../../helpers/user-provided-git'
+import {buildPath} from '../../helpers/utils'
+
+import {apiConstructor} from './api'
+import {APIHelper, Payload} from './interfaces'
+import sarifJsonSchema from './json-schema/sarif-schema-2.1.0.json'
+import {
+  renderCommandInfo,
+  renderSuccessfulCommand,
+  renderDryRunUpload,
+  renderRetriedUpload,
+  renderFailedUpload,
+  renderInvalidFile,
+} from './renderer'
+import {getBaseIntakeUrl} from './utils'
+
+const errorCodesStopUpload = [400, 403]
+
+const ajv = new Ajv()
+addFormats(ajv)
+
+const validateSarif = (sarifReportPath: string) => {
+  const sarifJsonSchemaValidate = ajv.compile(sarifJsonSchema)
+  try {
+    const sarifReportContent = JSON.parse(String(fs.readFileSync(sarifReportPath)))
+    const valid = sarifJsonSchemaValidate(sarifReportContent)
+    if (!valid) {
+      return sarifJsonSchemaValidate.errors
+    }
+  } catch (error) {
+    return error.message
+  }
```

Comment:
you're returning `.errors` or `error.message`: are they of the same type?

Reply:
Changed. Now it should return a `string`. However, not sure if that's the correct way to represent the array as `string`.